### ISSUE TITLE
performance after > 30 epochs

### ISSUE THREAD

**Issue by mikerossgithub** [XifengGuo/CapsNet-Keras#16]
I tried running for 200 epochs a few times. The performance stabilizes somewhere in the 30-50 epoch range. The paper shows 1250 epochs so something is different. I can't reproduce the result either. Median error with 3 routing iterations and reconstruction loss is 0.36, far outside of the 0.25+-0.005 reported. Maybe this is because the learning rate decays too fast? I think the more interesting result in the paper is the robustness to affine transformation. But its odd that the raw reported mnist error cant be reproduced.

**Comment by XifengGuo** [comment]
[@mikerossgithub](https://github.com/mikerossgithub) Yeah, I do believe that something is different from the paper. I can’t get the error of 0.25 either using the current version. But I haven’t figured out the reason and I haven’t seen the exact implementation. I’ll keep studying on this. Congratulations are welcome!

**Comment by sulaimanvesal** [comment]
@XifengGuo Thank you for your nice script, I ran your code with a bit changes, and here is my output, after 17 epochs:

```
Epoch 17/50
599/600 [============================>.] - ETA: 0s - loss: 0.0136 - capsnet_loss: 0.0043 - decoder_loss: 0.0238 - capsnet_acc: 0.9972Epoch 00016: val_capsnet_acc improved from 0.99540 to 0.99600, 
600/600 [==============================] - 175s - loss: 0.0136 - capsnet_loss: 0.0043 - decoder_loss: 0.0238 - capsnet_acc: 0.9972 - val_loss: 0.0129 - val_capsnet_loss: 0.0047 - val_decoder_loss: 0.0209 - val_capsnet_acc: 0.9960
```
It seems the model worked well, at least I am saying based on Val_capsnet_acc ? Am I right!

**Comment by XifengGuo** [comment]
[@sulaimanvesal](https://github.com/sulaimanvesal) Yeah, It works fine. Feel free to try on new tasks.
